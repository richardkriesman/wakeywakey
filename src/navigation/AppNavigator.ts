/**
 * @module navigation
 */

import { createAppContainer } from "react-navigation";

import MainStackNavigator from "./MainStackNavigator";

/**
 * Default app container. Connects the MainStackNavigator.
 * @author Shawn Lutch, Miika Raina
 */
export default createAppContainer(MainStackNavigator);
