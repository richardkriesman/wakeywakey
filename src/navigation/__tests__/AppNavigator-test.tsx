import MockDate from "mockdate";
import * as React from "react";

import NavigationTestUtils from "react-navigation/NavigationTestUtils";
import Renderer from "react-test-renderer";
import AppNavigator from "../AppNavigator";

const fakeDateMillis = 1551896555862;

describe("AppNavigator", () => {
    jest.useFakeTimers();

    beforeEach(() => {
        NavigationTestUtils.resetInternalState();
        MockDate.set(fakeDateMillis);
    });

    it("renders properly", () => {
        const tree = Renderer.create(<AppNavigator/>);
        expect(tree.toJSON()).toMatchSnapshot();
    });
});
