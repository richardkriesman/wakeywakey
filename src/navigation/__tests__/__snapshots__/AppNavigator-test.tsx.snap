// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    enabled={false}
    hitSlop={
      Object {
        "right": 50,
      }
    }
    maxDeltaY={20}
    minOffsetX={5}
    onGestureEvent={[Function]}
    onHandlerStateChange={[Function]}
  >
    <View
      style={
        Object {
          "backgroundColor": "#eee",
          "flex": 1,
          "flexDirection": "column-reverse",
          "overflow": "hidden",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          pointerEvents="auto"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": 0,
                },
              ],
            }
          }
        >
          <View
            accessibilityElementsHidden={false}
            style={
              Object {
                "backgroundColor": "#fff",
                "flex": 1,
              }
            }
          >
            <RCTSafeAreaView
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                onLayout={[Function]}
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <View
                  onMoveShouldSetResponder={[Function]}
                  onMoveShouldSetResponderCapture={[Function]}
                  onResponderEnd={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderReject={[Function]}
                  onResponderRelease={[Function]}
                  onResponderStart={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  onStartShouldSetResponderCapture={[Function]}
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                >
                  <View
                    accessible={true}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "marginTop": 20,
                          "padding": 0,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "flex": 1,
                            "flexBasis": "100%",
                            "justifyContent": "center",
                            "width": "85%",
                          }
                        }
                      >
                        <View
                          style={Object {}}
                        >
                          <View
                            style={
                              Object {
                                "marginBottom": 20,
                                "marginTop": 20,
                              }
                            }
                          >
                            <Text
                              style={
                                Object {
                                  "fontSize": 60,
                                  "textAlign": "center",
                                }
                              }
                            >
                              12:22 pm
                            </Text>
                          </View>
                        </View>
                        <View
                          style={
                            Object {
                              "borderRadius": 3,
                            }
                          }
                        >
                          <View
                            accessible={true}
                            isTVSelectable={true}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "opacity": 1,
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "alignItems": "center",
                                  "backgroundColor": "#000000",
                                  "borderColor": "#2089dc",
                                  "borderRadius": 3,
                                  "borderWidth": 0,
                                  "flexDirection": "row",
                                  "justifyContent": "center",
                                  "padding": 15,
                                }
                              }
                            >
                              <Text
                                style={
                                  Object {
                                    "color": "white",
                                    "fontSize": 18,
                                    "paddingBottom": 1,
                                    "paddingTop": 2,
                                    "textAlign": "center",
                                  }
                                }
                              >
                                Snooze
                              </Text>
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </RCTSafeAreaView>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#FFF",
              "transform": Array [
                Object {
                  "translateX": -750,
                },
              ],
            }
          }
        >
          <View
            onLayout={[Function]}
            pointerEvents="box-none"
            style={
              Object {
                "backgroundColor": "#FFF",
                "borderBottomColor": "#A7A7AA",
                "borderBottomWidth": 0.5,
                "height": 64,
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 20,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
